framework/src/actions/CreateExecutionerAction.C

```diff
@@ -137,11 +134,17 @@ CreateExecutionerAction::act()
     _problem->getNonlinearSystem()._compute_initial_residual_before_preset_bcs = getParam<bool>("compute_initial_residual_before_preset_bcs");
   }
 
+  Moose::setup_perf_log.push("Create Executioner","Setup");
+  _moose_object_pars.set<FEProblem *>("_fe_problem") = _problem.get();
+  MooseSharedPointer<Executioner> executioner = MooseSharedNamespace::static_pointer_cast<Executioner>(_factory.create(_type, "Executioner", _moose_object_pars));
+  Moose::setup_perf_log.pop("Create Executioner","Setup");
+
   _awh.executioner() = executioner;
 }
 
+#ifdef LIBMESH_HAVE_PETSC
 void
-CreateExecutionerAction::storeCommonExecutionerParams(FEProblem & fe_problem, InputParameters & params)
+CreateExecutionerAction::storePetscOptions(FEProblem & fe_problem, const InputParameters & params)
```

Comment:
Hmm I'm not sure if this name is quite right. It's fine that you appended the body of `FEProblem::storePetscOptions()` to this utility routine in CreateExecutionerAction, but the intent was that this routine handled Executioner options and handled the PetscOptions. It appears that we don't really have special Executioner options handled here so we could just deal with them in act() instead.

If you are churning things around why not just move this whole routine to PetscSupport.C? This make more sense since we also call this routine from `SetupPreconditionerAction`. I'm not sure it's an improvement to move this method out of FEProblem since it forces that cross Action call.